### ISSUE TITLE
[Feature Request] Support "protocol" & "forward-addr" in the config file.

### ISSUE THREAD

**Issue by Tigerfyj** [shadowsocks/shadowsocks-rust#451]
I wrote a systemd file for sslocal on my local machine. If the config file can support all commmand flags, that would be easy to change my settings.

**Comment by database64128** [comment]
I'm pushing towards the same goal, starting with #440.

**Comment by zonyitoo** [comment]
How about this new design of `locals` configuration?

Posted in https://github.com/shadowsocks/shadowsocks-rust/issues/440#issuecomment-792004622

```jsonc
{
    "locals": [
        {
            "local_address": "127.0.0.1",
            "local_port": 1081
            // socks5 by default
        },
        {
            "local_address": "127.0.0.1",
            "local_port": 3128,
            "protocol": "http"
        },
        {
            "local_address": "0.0.0.0",
            "local_port": 60080,
            "protocol": "redir",
            "tcp_redir": "redirect",
            "udp_redir": "tproxy"
        },
        {
            "local_address": "::",
            "local_port": 60080,
            "protocol": "redir",
            "tcp_redir": "redirect",
            "udp_redir": "tproxy",
            "ipv6_only": true
        },
        {
            "local_address": "::",
            "local_port": 53,
            "protocol": "tunnel",
            "forward_address": "8.8.8.8",
            "forward_port": 53,
            "ipv6_only": false
        }
    ]
}
```

**Comment by zonyitoo** [comment]
Merged into master and will be released in v1.10.0.
